guide/src/tutorial.md

```diff
@@ -0,0 +1,442 @@
+not relative to the current directory.
+
+The first three files contain package-level configuration settings. You may already be familiar with
+`Cargo.toml` and `rust-toolchain`, generic files for Rust projects. The file `why3find.json` is
+used by a verification tool invoked by Creusot. We will leave these files alone for this tutorial.
+
+The last file, `src/lib.rs`, is what we will be editing in this tutorial.
+
+## My first contract
+
+Open `src/lib.rs`. It contains an example function with a *contract*:
+
+```rust
+use creusot_contracts::*;
+
+#[requires(x@ < i64::MAX@)]
+#[ensures(result@ == x@ + 1)]
+pub fn add_one(x: i64) -> i64 {
+    x + 1
+}
+```
+
+In Creusot, contracts are how we specify the expected behavior of functions.
+A contract contains two kinds of assertions: *preconditions*, or "required clauses",
+describe the values of arguments that the function accepts;
+*postconditions*, or "ensured clauses", describe the result value that the function returns.
+
+In this first example, there is one precondition (`requires`) and one postcondition (`ensures`).
+This contract can be read as follows: if the argument `x` is smaller than `i64::MAX`,
+then the value of `add_one(x)` will be equal to `x@ + 1`.
+
+More generally, a function *satisfies its contract* when, for any input arguments that satisfy
+the preconditions (`requires`), the result of the function satisfies the postcondition (`ensures`).
+
+It is not a very exciting contract, since it just replicates the body of the function `x + 1`.
+But note that `x@ + 1` is not exactly the same expression. The `@` suffix is Creusot's *view operator*,
+which maps a Rust value to a mathematical value. If `x` has type `i64`, then `x@` has type `Int`,
+as the type of mathematical integers (..., -2, -1, 0, 1, 2, ...).
+The key difference is that mathematical integers are unbounded: their arithmetic operations do not overflow.
+They are more intuitive and better behaved than operations on machine integer.
+For example, all `x: Int` satisfy `x < x + 1` whereas this cannot be true for Rust's machine integers because they are finite.
+
+The type `Int` is one of several built-in types of common mathematical objects defined by Creusot;
+others include sequences and mappings (mathematical functions).
+On the one hand, mathematical objects like `Int` cannot be manipulated by Rust programs, they can only be used in contracts.
+On the other hand, values that exist in Rust programs such as machine integers (`i64`, etc.) are not as easy to manipulate
+in the logical world of contracts, so we use the view operator `@` to transform them into something more intuitive.
+
+The contract of `add_one` specifies the result slightly indirectly: its view as an unbounded integer, denoted by `result@`,
+must be equal to `x@ + 1`. For technical reasons, arithmetic operations are only defined on `Int` in the logical world of contracts.
```

Comment:
I think we can drop 'For technical reasons' (Or alternatively, explain what those reasons are)